rllib/algorithms/impala/impala.py

```diff
@@ -566,18 +564,17 @@ def setup(self, config: AlgorithmConfig):
             i: [] for i in range(self.config.num_learners or 1)
         }
 
-        # Create our local mixin buffer if the num of aggregation workers is 0.
+        # Create our local mixin buffer.
         if not self.config.enable_rl_module_and_learner:
-            if self.config.replay_proportion > 0.0:
-                self.local_mixin_buffer = MixInMultiAgentReplayBuffer(
-                    capacity=(
-                        self.config.replay_buffer_num_slots
-                        if self.config.replay_buffer_num_slots > 0
-                        else 1
-                    ),
-                    replay_ratio=self.config.replay_ratio,
-                    replay_mode=ReplayMode.LOCKSTEP,
-                )
+            self.local_mixin_buffer = MixInMultiAgentReplayBuffer(
```

Comment:
Dumb question: why do we even have a "mixin" replay buffer. Couldn't we just use a normal one and mix the batch in by ratio?

Reply:
We could, but then again: This is old API stack. Not the time for any changes or experiments ;)